src/entityManager.ts

```diff
@@ -0,0 +1,88 @@
+import { Characters } from './constants';
+import { Entity } from './entities/entity';
+import { PlayerEntity } from './entities/player.entity';
+import { RemoveEntityComponent } from './components/removeEntity.component';
+
+/**
+ * Manages entities within the game.
+ * @class
+ */
+export class EntityManager {
+  private entities: Entity[] = [];
```

Comment:
Your suggestion makes sense, and I appreciate the input. I considered using Set, but since I'm currently working in an ES5 environment, I'm unable to easily iterate through Sets. Considering the quicker dictionary operations that Maps or Sets offer, upgrading to ES2015 to gain access to Set iteration sounds like a reasonable approach. I'll explore that option further to see how it fits into the overall architecture of the project.

Reply:
Should I just move to ES2015? for-of does not work for Set in ES5.

Reply:
Any reason not to use the latest standard?

Reply:
After conducting a brief analysis, I've decided to upgrade to ES6. In case there's a need to revert to a lower version, there are tools available for that purpose.

Reply:
`entities` now stored in a Set 49412f5b90b1f281b0a376eb57aabf2e108a7842